File: Code Snippets/capture-latency.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <chrono>

using namespace std::chrono;
using namespace cv;
using namespace std;

int main() {
    Mat frame;
    namedWindow("Webcam Output");

    VideoCapture cap(0);

    if (!cap.isOpened()) {
        cout << "webcam in use";
    }

    time_t start, end;
    int amount_of_frames = 1000;
    double sum_latency = 0;
    for (size_t i = 0; i < 1000; i++) {
        auto start = high_resolution_clock::now();
        cap >> frame;

        imshow("Webcam Output", frame);
        waitKey(1);
        auto stop = high_resolution_clock::now();
        sum_latency += duration_cast<milliseconds>(stop - start).count;
    }

    cout << "Average Latency: " << sum_latency / amount_of_frames << setprecision(5) << " msec" <<endl;

    return 0;
}

File: Code Snippets/capture-latency.py
import cv2
import time
stream = cv2.VideoCapture(0,cv2.CAP_DSHOW)

start_interval=0
sum_latency=0
frames=0
while(True):
    start_interval=time.time()
    ret, frame = stream.read()
    if not ret:
        #Webcam disconnect or used by another application
        break

    cv2.imshow("Webcam_Output", frame)

    latency = time.time() - start_interval
    sum_latency=sum_latency+latency
    frames=frames+1
    
    pressed_key=cv2.waitKey(1)
    if pressed_key == ord('q'):
        break

stream.release()
cv2.destroyAllWindows

print(f"Average Latency: {sum_latency/frames}")